sequencer/src/api/sql.rs

```diff
@@ -166,19 +167,51 @@ impl CatchupStorage for SqlStorage {
             .read()
             .await
             .context(format!("opening transaction to fetch leaf at {height}"))?;
-        let h = usize::try_from(height)?;
-        let query_leaf_chain = tx
-            .get_leaf_range(h..=(h + 2))
+        let leaf = tx
+            .get_leaf((height as usize).into())
             .await
-            .context(format!("leaf chain {height} not available"))?;
-        let mut chain = vec![];
+            .context(format!("leaf {height} not available"))?;
+        let mut last_leaf: Leaf2 = leaf.leaf().clone();
+        let mut chain = vec![last_leaf.clone()];
+        let mut h = height + 1;
+
+        let mut found = false;
```

Comment:
Nit: we don't really need this variable, we always break right after setting it true